File: backend/Entities/Message.cs
namespace backend.Entities
{
    public class Message
    {
        public int Id { get; set; }
        public string UserEmail { get; set; } = string.Empty;
        public string Content { get; set; } = string.Empty;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public string UserName { get; set; } = string.Empty;
    }
}

File: frontend/src/Models/Message.ts
// Purpose: Interface for Message object.
export interface MessageRequest {
  type: string;
  userName: string;
  content: string;
  createdAt: Date;
}

// Purpose: Interface for Message object.
export interface MessageResponse {
  Id: number;
  userEmail: string;
  content: string;
  createdAt: Date;
  userName: string;
}

// Purpose: Interface for UserCountMessage object.
export interface UserCountMessage {
  type: string;
  count: number;
}
